### DISCUSSION TITLE
How to chain filters similar to logical AND?

### DISCUSSION THREAD

**Discussion by dayemsaeed** [supabase/supabase#28016] [Questions] [answered]
I'm trying to fetch data within a range of timestamps e.g: "2024-07-16 00:00:00+00" and "2024-07-16 23:59:59+00". I'm fetching the data like so:
`val response = supabaseClient
                    .from("tasks")
                    .select {
                        filter {
                            gte("task_created_at", startOfDayUTC)
                            lte("task_created_at", endOfDayUTC)
                            eq("task_created_by", userId)
                        }
                        order("task_created_at", Order.ASCENDING)
                    }`

But from my understanding, the chaining of filters is a logical "OR". So if I have a task on 2024-07-14, it will show up as well since it fulfills the `lte("task_created_at", endOfDayUTC)` condition since the 14th comes before the 16th

How can I chain it so that it is a logical AND and gets only the timestamps within the range I gave. I can't use `rangeGte` or similar filters since my column type is "Timestamptz" not "tsrange"

**Comment by GaryAustin1** [comment, marked as answer]
Chaining of filters is logical AND.  So you can do .select().gte().lte() to find values between.

**Reply by GaryAustin1** [reply]
![image](https://github.com/user-attachments/assets/be81bfd5-c12f-4be1-bc2b-93df2c7e43f3)
https://supabase.com/docs/reference/javascript/using-filters
